lib/bandit/headers.ex

```diff
@@ -67,6 +64,29 @@ defmodule Bandit.Headers do
     end
   end
 
+  # Parses non-negative integers from strings. Return the valid portion of an
+  # integer and the remaining string as a tuple like `{123, "")}` or `:error`.
+  def parse_integer(binary) do
+    parse_integer(binary, [])
```

Comment:
Instead of accumulating a list can you just accumulate the number directly? Start with zero in the accumulator, and on every digit read set the accumulator to `(accumulator * 10) + new_digit`?